credential/util.go

```diff
@@ -0,0 +1,128 @@
+package credential
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/goccy/go-json"
+	"github.com/lestrrat-go/jwx/v2/jws"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+	"github.com/pkg/errors"
+)
+
+// ToCredential turn a generic cred into its known object model
+func ToCredential(genericCred any) (jws.Headers, jwt.Token, *VerifiableCredential, error) {
```

Comment:
A lot of code reuse, would it be possible to Call this funciton in the signature.go VerifyCredentialSignature?

Reply:
good idea, I think it can be used for everything but the JWT case